src/main/java/com/es/phoneshop/model/product/Product.java

```diff
@@ -13,76 +21,11 @@
     private Currency currency;
     private int stock;
     private String imageUrl;
-    public Product(Long id, String code, String description,
-                   BigDecimal price, Currency currency, int stock, String imageUrl) {
-        this.id = id;
-        this.code = code;
-        this.description = description;
-        this.price = price;
-        this.currency = currency;
-        this.stock = stock;
-        this.imageUrl = imageUrl;
-    }
+    private ArrayList<PriceHistory> priceHistories;
```

Comment:
https://medium.com/@sanchezdale/oop-good-practices-coding-to-the-interface-baea84fd60d3